waspc/cli/Command/BashCompletion.hs

```diff
@@ -6,22 +6,32 @@ module Command.BashCompletion
 where
 
 import Command (Command)
+import Control.Exception (assert)
 import Control.Monad.IO.Class (liftIO)
+import Data.List (isPrefixOf)
 import Paths_waspc (getDataFileName)
+import qualified System.Environment as ENV
 
-bashCompletion :: [String] -> Command ()
-bashCompletion args = case args of
-  [] -> listCommands
-  ["db"] -> listCommandsDb
-  _ -> liftIO . putStrLn $ ""
-
--- return available options for wasp <command>
-listCommands :: Command ()
-listCommands = liftIO . putStrLn $ unlines ["new", "version", "start", "db", "clean", "build", "telemetry", "deps"]
-
--- return available options for wasp db <command>
-listCommandsDb :: Command ()
-listCommandsDb = liftIO . putStrLn $ unlines ["migrate-dev", "studio"]
+-- generate bash completion depending on commands input
+bashCompletion :: Command ()
+bashCompletion = do
+  -- COMP_LINE is exposed by the bash `complete` builtin (https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html)
+  inputEntered <- liftIO (ENV.getEnv "COMP_LINE")
+  let inputWords = words inputEntered
+  let inputArgs = assert (head inputArgs == "wasp") $ tail inputWords
+  case inputArgs of
+    [] -> listCommands commands
+    ["db"] -> listCommands dbSubCommands
+    [cmdPrefix] -> listMatchingCommands cmdPrefix commands
+    ("db" : [cmdPrefix]) -> listMatchingCommands cmdPrefix dbSubCommands
```

Comment:
This is ok but I am curious, why not `["db", cmdPrefix]` instead of `("db" : [cmdPrefix])`?

Reply:
done